### DISCUSSION TITLE
Lesson 30: don't you think you should so some check in `fulfillRandomWords`?

### DISCUSSION THREAD

**Discussion by fabriziogianni7** [Cyfrin/foundry-full-course-cu#1537] [General]
# Description

don't you think you should so some check in `fulfillRandomWords`? I think we should check if the request id is the current one generated in `performUpkeep` function and that the caller is the coordinator contract.

```
 function fulfillRandomWords(
        uint256 requestId,
        uint256[] memory randomWords
    ) internal override {
        if (requestId != s_requestId) revert Raffle__WrongRequestId(requestId);
        if (msg.sender != vrf.vrfCooridnatorAddress)
            revert Raffle__OnlyCoordinatorCanCallFulfillRandomWords(msg.sender);
            // rest of the code
```

Why in the course, Patrick does not add this chex?

**Comment by EngrPips** [comment]
Hello @fabriziogianni7 , Tutorial code are always more loose than production code and also the instructor sometimes would intentionally make it loose so that the students can have something to improve on. It would be a great learning experience if you implement those checks
